### PR TITLE
New token protos, format, APIs

### PR DESCRIPTION
**Pull request by mitchellh**
This PR adds the new token internal format and APIs. This doesn't affect any user-visible portion of Waypoint yet and is fully backwards compatible. This is laying some groundwork for the upcoming user system and OIDC integrations.

I was originally going to add the `waypoint user` CLI to this but I thought even this was a good PR step before I did that since it is fairly isolated as-is and the backwards compatibility is shown well by keeping `waypoint token` around.

No changelog entry on this because I think this will be rolled into the OIDC changelog entry.

### CODE REVIEW

**Review by krantzinator — APPROVED** [review]
Looks like a good base!
As an aside, are you running something outside of `make format` to trim the line-ending spaces in the proto file? Curious since it looks like it caught a bunch of my comments 😄

**Review by evanphx — APPROVED** [review]
Looks good! This is a great example PR for how to upgrade a protobuf with new fields and new functionality.

File: internal/server/singleprocess/auth.go [hunk 1]
```diff
@@ -176,118 +226,187 @@ func (s *service) GenerateToken(keyId string, metadata map[string]string, body *
...
+
+	if req.Login == nil {
+		req.Login = &pb.Token_Login{
+			// TODO(mitchellh): when we have a login system, set this
+			// to the currently logged in user.
+			UserId: DefaultUser,
+		}
+	}
+
+	// We don't currently have any other users.
```
[the remark this answers is not in the document]
Comment (mitchellh) [reply internal/server/singleprocess/auth.go:315]: I think I'll get to it pretty fast cause tests will fail and its what I'm working on next :)

**Comment by mitchellh** [comment]
> As an aside, are you running something outside of `make format` to trim the line-ending spaces in the proto file? Curious since it looks like it caught a bunch of my comments 😄

I have an editor hook that trims trailing spaces on all lines of all files haha.

**Comment by mitchellh** [comment]
Will merge this given two 👍 . We aren't releasing for awhile so if there is any further feedback happy to retroactively apply it.

**Pull request merged**